subprojects/core/src/main/java/org/gradle/api/tasks/Sync.java

```diff
@@ -69,4 +86,27 @@ public void setDestinationDir(File destinationDir) {
         into(destinationDir);
     }
 
+    /**
+    * Returns the CopySpec that defines what files to preserver in {@link #getDestinationDir()}
+    *
+    * @return    the CopySpec for files to preserve
+    *
+    * @see #getDestinationDir()
+    */
+    @Internal
+    public CopySpec getPreserve() {
+        return preserveInDestination;
+    }
+
+    /**
+    * Configures the patterns to preserve in the destination directory.
+    *
+    * @param configureClosure
+    *
+    * @see #getDestinationDir()
+    */
+    public void preserve(Closure configureClosure) {
```

Comment:
Please add only an Action based method. The Closure based method will be added by runtime decoration. The Signature should be
```java
public void preserve(Action<CopySpec> action)
```

You can have a look at `CopySpec`, there are various examples of methods like that.